web/src/settings_ui.ts

```diff
@@ -6,46 +6,63 @@ import {$t, $t_html} from "./i18n";
 import * as loading from "./loading";
 import * as ui_report from "./ui_report";
 
-export function display_checkmark($elem) {
+export function display_checkmark($elem: JQuery<HTMLElement>): void {
     const check_mark = document.createElement("img");
     check_mark.src = checkbox_image;
     $elem.prepend(check_mark);
     $(check_mark).css("width", "13px");
 }
 
-export const strings = {
+type Strings = {
+    failure_html: string;
+    saving: string;
+    success_html: string;
+};
+
+export const strings: Strings = {
     success_html: $t_html({defaultMessage: "Saved"}),
     failure_html: $t_html({defaultMessage: "Save failed"}),
     saving: $t({defaultMessage: "Saving"}),
 };
 
+type DoSettingsChangeOptions = {
+    failure_msg_html?: string;
+    sticky?: boolean;
+    success_msg_html?: string;
+    $error_msg_element?: JQuery<HTMLElement>;
+    error_continuation?(xhr?: JQuery.jqXHR): void;
+    success_continuation?(response_data?: unknown): void;
```

Comment:
The `xhr` and `response_data` parameters should not be marked optional.

Reply:
@andersk I checked the places where `do_settings_change` is used, and they're not providing any arguments to `success_continutation` and `error_continutation`.

Here I'm linking the places:-
https://github.com/zulip/zulip/blob/ebc07c0dd7a4de98c58565ccf85f83e4bb2f10a4/web/src/settings_account.js#L646
https://github.com/zulip/zulip/blob/ebc07c0dd7a4de98c58565ccf85f83e4bb2f10a4/web/src/settings_account.js#L690

Reply:
The *caller* of `success` or `error` always passes it an argument. It is up to the function whether or not to bind it as a parameter. Since *some* of these functions will bind it as a parameter, the type annotation needs to reflect the fact that the caller *always* passes the argument.

Reply:
Fixed.